templates/Makefile

```diff
@@ -1,36 +1,34 @@
+PLATFORM = $(shell uname -m)
+
+ifneq ($(PLATFORM),aarch64)
 ifeq ($(origin AS),default)
 AS = aarch64-linux-gnu-as
 endif
 ifeq ($(origin CC),default)
 CC = aarch64-linux-gnu-gcc
 endif
+endif
 
-CFLAGS ?= -g -Wall -Wextra -pedantic -Werror
-LDFLAGS =
-
-ALL_LDFLAGS = -pie -Wl,--fatal-warnings
-
-ALL_CFLAGS = -std=c99 -fPIE $(CFLAGS)
-ALL_LDFLAGS += $(LDFLAGS)
+CFLAGS ?= -g -Wall -Wextra -pedantic -Werror -std=c99 -fPIE 
+LDFLAGS ?= -pie -Wl,--fatal-warnings
```

Comment:
This change is somewhat unrelated, but having `LDFLAGS` and `ALL_LDFLAGS` didn't make that much sense to me. With just a single variable that is set via `?=` we give the user all options for overriding flags.